distributed/comm/tcp.py

```diff
@@ -260,6 +259,8 @@ async def write(self, msg, serializers=None, on_error="message"):
             for each_frame in frames:
                 each_frame_nbytes = nbytes(each_frame)
                 if each_frame_nbytes:
+                    if stream._write_buffer is None:
+                        raise StreamClosedError()
```

Comment:
Treat ``'NoneType' object has no attribute 'append'``; see handler below:
- ``if shutting_down()``, silently suppress the error
- otherwise, re-raise it as CommClosedError